plugin/trino-open-policy-agent/src/main/java/io/trino/plugin/openpolicyagent/schema/TrinoTable.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openpolicyagent.schema;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import com.google.common.collect.ImmutableSet;
+import io.trino.spi.connector.CatalogSchemaTableName;
+
+import java.util.Set;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public record TrinoTable(@JsonUnwrapped TrinoSchema catalogSchema,
+                         String tableName,
+                         Set<String> columns)
+{
+    public static class Builder
+            extends BaseSchemaBuilder<TrinoTable, Builder>
+    {
+        public String tableName;
+        public Set<String> columns;
+
+        @Override
+        protected Builder getInstance()
+        {
+            return this;
+        }
+
+        public Builder tableName(String tableName)
+        {
+            this.tableName = tableName;
+            return this;
+        }
+
+        public Builder columns(Set<String> columns)
+        {
+            this.columns = ImmutableSet.copyOf(columns);
+            return this;
+        }
+
+        public static Builder fromTrinoTable(CatalogSchemaTableName table)
+        {
+            return new Builder()
+                    .catalogName(table.getCatalogName())
+                    .schemaName(table.getSchemaTableName().getSchemaName())
+                    .tableName(table.getSchemaTableName().getTableName());
+        }
+
+        @Override
+        public TrinoTable build()
+        {
+            return new TrinoTable(this);
+        }
+    }
+
+    public TrinoTable(Builder builder)
+    {
+        this(new TrinoSchema(builder), builder.tableName, builder.columns);
+    }
+
+    public static TrinoTable fromTrinoTable(CatalogSchemaTableName table)
+    {
+        return TrinoTable.Builder.fromTrinoTable(table).build();
```

Comment:
I implemented the logic to create a Builder from the internal Trino classes because it simplifies the creation of these records when some of their fields will be populated from a Trino internal object but others will be populated from somewhere else - e.g. `checkCanSelectFromColumns`: the table information comes from `CatalogSchemaTableName` while the columns are coming from another argument as per the SPI. So while that method may look redundant it comes in handy sometimes.

Using it here makes it (IMHO) a bit cleaner than implementing the logic to call the constructor.

If I call the constructor directly here, I am duplicating the logic that is already in the Builder. It would look something like
```java
    public static TrinoTable fromTrinoTable(CatalogSchemaTableName table)
    {
        return new TrinoTable(
                TrinoSchema.builder()
                        .catalogName(table.getCatalogName())
                        .schemaName(table.getSchemaTableName().getSchemaName())
                        .build(),
                table.getSchemaTableName().getTableName(),
                null);
    }
```

Whereas relying on the `Builder` abstracts away the fact that a `TrinoTable` contains a `TrinoSchema`, as it's all automatically handled by the builders.

Happy to change it if that's more in line with the style, just thought I'd try to clarify first